client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -295,12 +295,16 @@ class ThemesMagicSearchCard extends React.Component {
 								/>
 							</div>
 						}
-						{ isPremiumThemesEnabled && ! isJetpack &&
-							<SegmentedControl
-								initialSelected={ this.props.tier }
-								options={ tiers }
-								onSelect={ this.props.select }
-							/>
+						{
+							(
+								( config.isEnabled( 'jetpack/pijp' ) && isPremiumThemesEnabled ) ||
+								( isPremiumThemesEnabled && ! isJetpack )
```

Comment:
I think you could simplify the condition to

```js
isPremiumThemesEnabled && ( ! isJetpack || config.isEnabled( 'jetpack/pijp' ) )
```

which I think also reads intuitive enough.